relay-server/src/utils/metrics_rate_limits.rs

```diff
@@ -36,18 +37,15 @@ pub struct MetricsLimiter<Q: AsRef<Vec<Quota>> = Vec<Quota>> {
 
 const PROFILE_TAG: &str = "has_profile";
 
-/// Extracts the transaction count from a metric.
+/// Extracts quota information from a metric.
 ///
 /// If the metric was extracted from a or multiple transaction, it returns the amount
 /// of datapoints contained in the bucket.
 ///
 /// Additionally tracks whether the transactions also contained profiling information.
 ///
 /// Returns `None` if the metric was not extracted from transactions.
-pub fn extract_transaction_count(
-    metric: &BucketView<'_>,
-    mode: ExtractionMode,
-) -> Option<TransactionCount> {
+fn count_metric_bucket(metric: BucketView<'_>, mode: ExtractionMode) -> Option<TransactionCount> {
```

Comment:
Should we get rid of `TransactionCount` and just return a `SourceQuantities` here (with `buckets` set to 1)? Or rather, unite this function with `extract_metric_quantities`. The function name already suggests it counts more than just transaction and profile info.

Reply:
Discussed with @Dav1dde and decided that it's useful to have both functions still. I'm open to more refactoring but since both of those are internal functions it's not a priority for me right now.

Reply:
Makes sense. I would personally rename it back to something like `get_transaction_count` then, but up to you.

Reply:
Agree with Joris here, I think the original name `extract_transaction_count` or `get_transaction_count` captures the intent of the function better.

We still need this behaviour for `MetricsLimiter` I believe (so can't just yet get rid of `TransactionCount` and combine it with `extract_metric_quantities`)